pkg/doc/verifiable/json.go

```diff
@@ -0,0 +1,82 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package verifiable
+
+import (
+	"encoding/json"
+)
+
+// marshalWithExtraFields marshals value merged with extra fields defined in the map into JSON.
+func marshalWithExtraFields(v interface{}, ef map[string]interface{}) ([]byte, error) {
+	// Convert value into a JSON map of known fields.
+	kf, err := mergeExtraFields(v, ef)
+	if err != nil {
+		return nil, err
+	}
+
+	// Marshal extended known fields map.
+	return json.Marshal(kf)
+}
+
+// unmarshalWithExtraFields unmarshals JSON into value and puts all JSON fields which do not belong to value
```

Comment:
"value" refers to "v" here, and "extra fields" is "ef"? Perhaps it could be clearer to just use the argument names in the comment instead.

Reply:
for clarity, I'll put **value v** and **extra fields map ef**.

Reply:
Even better. Thanks :)